lib/kcerts/ssh.go

```diff
@@ -345,11 +343,15 @@ func (a SSHAgent) AddCertificates(privateKey PrivateKey, publicKey ssh.PublicKey
 		return fmt.Errorf("certificate is already expired or invalid, not adding")
 	}
 
+	// BUG(INFRA-8631): Do not add LifetimeSecs to AddedKey because the lifetime of the cert
+	// is already embedded in the cert returned by the auth server. Adding LifetimeSecs
+	// again will cause the ssh-agent on Windows to experience an impersonation token error.
+	// This failure causes the ssh-agent on Windows to not respond back to enkit via RPC
+	// which triggers an obscure EOF error.
```

Comment:
Is this equally incorrect for Linux?

Reply:
On Linux, removing the `LifetimeSecs` field does not affect the cert timeout because the timeout is set by the auth server.